clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -644,10 +646,14 @@ private KafkaConsumer(ConsumerConfig config,
      */
     @Override
     public void subscribe(List<String> topics, ConsumerRebalanceListener listener) {
+        subscribeTopics(topics, listener, false);
+    }
+
+    private void subscribeTopics(List<String> topics, ConsumerRebalanceListener listener, boolean isSubscribedViaPatternSubscription) {
         acquire();
```

Comment:
It's a minor thing, but I would move the acquire/release to the public subscribe method.